constructor/constructor.py

```diff
@@ -0,0 +1,131 @@
+import torch
+import torch.nn as nn
+from torchvision import datasets, transforms
+import pandas as pd
+import io
+import json
+import torch
+from sklearn.model_selection import train_test_split
+import random
+import pandas
+import structures
+import ast
+import types
+import argparse
+
+activations = nn.ModuleDict([
+                ['prelu', nn.RReLU()],
+                ['lrelu', nn.LeakyReLU()],
+                ['relu', nn.ReLU()],
+                ['sigmoid', nn.Sigmoid()],
+                ['softmax', nn.Softmax()],
+                ['logsoftmax', nn.LogSoftmax()],
+                ['elu', nn.ELU()],
+    ])
+
+loss_functions = {'MSELoss': torch.nn.MSELoss(), 'CrossEntropy': torch.nn.CrossEntropyLoss()}
+
+class CustomDataset(torch.utils.data.Dataset):
+    def __init__(self, df, target_list):
+        self.df = df
+        self.vectors = df['vector'].values
+        self.targets = self.df[target_list].values
+
+    def __len__(self):
+        return len(self.vectors)
+
+    def __getitem__(self, index):
+        return torch.FloatTensor(self.vectors[index]), torch.FloatTensor(self.targets[index])
+
+
+class NeuralNetwork(nn.Module):
+    def __init__(self, info, type):
```

Comment:
Move the logic out of constructor, as it will be more complex; just set the fields here